be able to transfer karma/butts between names

himmelmaus opened the pull request:
as referenced in issue #42

Commit:
added transfer command, still need to fix the tokeniser

Commit:
accidentally imported something I didn't need

Commit:
switched tokenisation to use regex, works for terms with spaces in them

neanias commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,50 @@
+        else:
+            string = ""
+            strings = []
+            for x in args:
```
`x` is not a clear variable name.

neanias commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,50 @@
+            strings = []
+            for x in args:
+                string = string + " " + x
+            for y in re.findall("(?![\'\"])[a-z,\s]+(?![\"\`])", string):
```
Nor is `y`.

neanias commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,50 @@
+
+        if len(args) == 2:
+            oldname = args[0].lower()
+            newname = args[1].lower()
```
These should both be underscored: `old_name`, `new_name`.

neanias commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,50 @@
+            string = ""
+            strings = []
+            for x in args:
+                string = string + " " + x
```
It would probably be cleaner to use a format string here.

himmelmaus committed:
change empty string check to not use re

himmelmaus committed:
keeping travis happy

himmelmaus committed:
stylistic changes and add confirmation message

code changes to meet @neanias's requests and adding a confirmation message of changed butt and karma values

neanias commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,52 @@
+        redis.hset(bot.config['System']['redis_prefix'] + "buttmaster", new_name, new_butts + old_butts)
+        redis.hdel(bot.config['System']['redis_prefix'] + "buttmaster", old_name)
+        
+        bot.message(channel, "{} now has {} karma and {} butts, {} now has no karma and no butts".format(new_name,old_karma+new_karma,old_butts+new_butts,old_name))
```
This line is **way** too long

himmelmaus replied:
line 49 I presume?

neanias replied:
Yes

himmelmaus replied:
go big or go home

himmelmaus replied:
but how am I meant I fix that? just splitting the arguments across several lines?

neanias replied:
Tell that to the linter

neanias replied:
Yes

neanias commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,52 @@
+            string = ""
+            strings = []
+            for word in args:
+                string = "{} {}".format(string, word)
```
What are you doing here, exactly? Are you just joining all the `args` together?

himmelmaus replied:
yes, so the regex can deal with it

neanias replied:
Python has a `join` function for lists to join them together into strings. I'd recommand changing this to `strings = args.join(" ")`.

himmelmaus replied:
oh yeah it does doesn't it, ok I'll add that

neanias reviewed (changes requested):
More comments. You're getting there, keep going!

himmelmaus committed:
use slightly nicer regex

himmelmaus committed:
stylistic changes to pass linter

himmelmaus committed:
make args concatenation less verbose

himmelmaus committed:
more stylistic changes for travis

himmelmaus commented:
@neanias are you happy with that in its current state?

himmelmaus committed:
Changes to permissions and error handling

Change it so people are able to transfer their own karma and butts to somebody else and add an error message when not enough arguments are given

Commit:
boop

Commit:
remove self karma permission, keeping travis happy

hreeder commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,55 @@
+            bot.message(channel, "Oops! Looks like you didn't give enough arguments!")
+            return
+
+    if sender.lower() == bot.config['System']['owner']:
```
IMO, this should come earlier - why perform the searching if the user isn't authorized to run the command?

himmelmaus replied:
oh, I had that originally but changed it for when I added the permission to be able to transfer your own karma

hreeder commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,55 @@
+            first_name = strings[0].lower()
+            second_name = strings[1].lower()
+        except IndexError:
+            bot.message(channel, "Oops! Looks like you didn't give enough arguments!")
```
Maybe echo the usage here to prompt the user as to how it should look?

himmelmaus replied:
will do

hreeder commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,55 @@
+
+@bot.command('transfer')
+def transfer(bot, channel, sender, args):
+    """"command to transfer karma & butts from one term to another"""
```
Docstring should contain usage for something like this

hreeder commented on ircbot/plugins/transfer.py:
```diff
@@ -0,0 +1,55 @@
+            first_name = args[0].lower()
+            second_name = args[1].lower()
+    else:
+        string = args.join(" ")
```
Iirc this should be `" ".join(args)`

neanias replied:
(sorry, my bad, I forgot how to Python)

Commit:
minor changes requested by tbrb

Commit:
change docstring and error message to use bot.trigger
